internal/services/controllers/jobs/controller.go

```diff
@@ -750,6 +750,18 @@ func (ec *JobsControllerImpl) queueStepRun(ctx context.Context, tenantId, stepId
 	// If the step run input is not set, then we should set it. This will be set upstream if we've rerun
 	// the step run manually with new inputs. It will not be set when the step is automatically queued.
 	if in := data.Input; len(in) == 0 || string(in) == "{}" {
```

Comment:
We should double check that a workflow run which gets replayed and throws new errors adds the new errors to the input data, not the old errors. I believe we clear all step run inputs on a replay, but would be good to double check!

Reply:
So, it looks like this maybe kinda works? Is it possible that on replay the error is updated in the db with the new one? It looks like what's happening is the first run throws an error and that's passed along to the `input` correctly, then when I replay the workflow, _all_ of the past run inputs get overwritten with the new error. Is that how it's supposed to happen?

Tested this by printing out a timestamp in the error message, and then on replay that timestamp would update in the original run (in addition to being updated in the replayed run)

Reply:
Yes, that's the correct behavior, if you're saying both the step run and the on failure step run are getting the new timestamp.

Reply:
yep 👍